cns/ipampool/monitor.go

```diff
@@ -101,6 +103,16 @@ func (pm *Monitor) Start(ctx context.Context) error {
 			subnetCIDR = nnc.Status.NetworkContainers[0].SubnetAddressSpace
 			subnetARMID = GenerateARMID(&nnc.Status.NetworkContainers[0])
 
+			pm.metastate.primaryIPAddresses = make(map[string]struct{})
+			// Add Primary IP to Map, if not present.
+			// This is only for Swift i.e. if NC Type is vnet.
+			for i := 0; i < len(nnc.Status.NetworkContainers); i++ {
```

Comment:
I would suggest using range instead, like: `for _, nc := range nnc.Status.NetworkContainers {...}

Reply:
that makes a copy of the NC, which we would need to update it and write it back in to the status. indexing makes more sense here. (we could `for i := range ...`, because it's a little prettier, but the C style is fine too)